dev-db/postgresql/files/postgresql.init-9.3-r1

```diff
@@ -144,14 +173,32 @@ description_reload="Simply sends the postgres process a SIGHUP signal, causing
            require a complete restart to take effect."
 reload() {
     ebegin "Reloading PostgreSQL @SLOT@ configuration"
-    su - postgres -c "${PG_CTL} reload -s -D ${DATA_DIR}"
+    start-stop-daemon \
+        --signal SIGHUP \
+        --user postgres:postgres \
+        --pidfile "${PIDFILE}"
     eend $?
 }
 
 description_promote="If the server is in standby, it is commanded to exit
             recovery and begin read-write operations."
 promote() {
     ebegin "Promoting PostgreSQL @SLOT@"
-    su - postgres -c "${PG_CTL} promote -s -D ${DATA_DIR}"
-    eend $?
+    local retval=1
+    if "${PG_CONTROLDATA}" -D "${DATA_DIR}" 2>/dev/null \
+        | grep 'in archive recovery' >/dev/null 2>&1; then
+        local promotefile="${DATA_DIR%/}/promote"
+        checkpath -q -f -m 644 -o postgres:postgres "${promotefile}" && \
```

Comment:
I can't find anything in the docs stating we need to create an empty file, and the source seems to indicate that the backend will create and remove it. Was this not your experience?

Reply:
Yeah, I can't find any related docs either. However, the source code of pg_ctl I checked is also a signal sent to postgresql after creating an empty `promote` file. If the promote succeeds, the newly created file will be automatically deleted, and if it fails, it will be kept.
https://github.com/postgres/postgres/blob/75bd846b68b09f33a57d9de96064ded4d392ca05/src/bin/pg_ctl/pg_ctl.c#L1207-L1231

And I have done a test of only sending a signal without manually creating the promote file, and it is confirmed that it is invalid.

You can reproduce it with following steps:

1. stop postgresql-14
2. touch `/var/lib/postgresql/14/data/standby.signal`
3. start postgresql-14
4. run `pg_controldata -D /var/lib/postgresql/14/data | grep 'Database cluster state'`
    - and it will appear as: `in archive recovery`
5. change this initd script to remove this promote creating line, and run `rc-service postgresql-14 promote`
6. executes step 4 again, and it still appears as: `in archive recovery`